spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,669 @@
+        df_metrics_PAM50 = pd.read_csv(fname_metrics_PAM50).astype({metric: float})
+        # Average metrics of healthy controls
+        df_avg_HC = average_hc(path_ref, metric, list_HC)
+
+        # Step 3. Determine compressed levels for both subject and PAM50 space
+        # --------------------------------------------------------------------
+        # Get slices correspondance in PAM50 space
+        compressed_levels_dict = get_slices_in_PAM50(compressed_levels_dict, df_metrics, df_metrics_PAM50)
+        z_range_below, z_range_above = get_slices_upper_lower_level_from_PAM50(compressed_levels_dict, df_metrics_PAM50, distance, extent, slice_thickness_PAM50)
+        # Get data from healthy control and average them
+    else:
+        # Get spinal cord centerline object to compute the distance
+        # Get max and min index of the segmentation with pmj
+        _, _, Z = (img_seg.data > NEAR_ZERO_THRESHOLD).nonzero()
+        min_z_index, max_z_index = min(Z), max(Z)
+        # Get the z index corresponding to the segmentation since the centerline only includes slices of the segmentation.
+        z_ref = np.array(range(min_z_index.astype(int), max_z_index.max().astype(int) + 1))
+        # Get centerline object
+        centerline = get_centerline_object(img_seg, verbose=verbose)
+        # Get healthy slices to average for level above and below
+        z_range_above, z_range_below = get_slices_upper_lower_level_from_centerline(centerline, distance, extent, slice_compressed, z_ref)
+
+    # Step 4. Compute MSCC metrics for each compressed level
+    # ------------------------------------------------------
+    # Loop through all compressed levels (compute one MSCC per compressed level)
+    for level in compressed_levels_dict.keys():
+        printv('\nLevel: {}'.format(level), verbose=verbose, type='info')
+        # Get metric of patient with compression
+        if arguments.normalize:
+            metrics_patient, slices_avg = average_compression_PAM50(slice_thickness, slice_thickness_PAM50, metric, df_metrics_PAM50,
+                                                                    z_range_above, z_range_below, compressed_levels_dict[level])
+            # Get metrics of healthy controls
+            metrics_HC = average_metric(df_avg_HC, metric, z_range_above, z_range_below, slices_avg)
+            logger.debug('\nmetric_a_HC =  {}, metric_b_HC = {}, metric_i_HC = {}'.format(metrics_HC[0], metrics_HC[1], metrics_HC[2]))
+            # Compute Normalized Ratio
+            metric_ratio_norm_result = metric_ratio_norm(metrics_patient, metrics_HC)
+        else:
+            slices_avg = compressed_levels_dict[level]
+            metrics_patient = average_metric(df_metrics, metric, z_range_above, z_range_below, slices_avg)
+            metric_ratio_norm_result = None
+        logger.debug('\nmetric_a =  {}, metric_b = {}, metric_i = {}'.format(metrics_patient[0], metrics_patient[1], metrics_patient[2]))
```

Comment:
I think this section would be a tiny bit easier to read if this `logger.debug` statement was moved to the same place as the `# Display results` section below. (Just to keep the three print-related statements together.)